src/libANGLE/renderer/d3d/DynamicHLSL.cpp

```diff
@@ -594,6 +613,30 @@ void DynamicHLSL::generateShaderLinkHLSL(const gl::Caps &caps,
         vertexGenerateOutput << "    output.gl_FragCoord = gl_Position;\n";
     }
 
+    const auto &usedClipDistances = programMetadata.usedClipDistances();
+    for (int i = 0; i < usedClipDistances.size(); i++)
+    {
+        const auto semanticIdx  = i / 4;
+        const auto clipPlaneIdx = usedClipDistances[i];
+        vertexGenerateOutput << "    output.gl_ClipDistance" << semanticIdx << '.';
+        switch (i % 4)
+        {
+            case 0:
+                vertexGenerateOutput << 'x';
+                break;
+            case 1:
```

Comment:
nit: you could use `op[]` on this vector instead of the switch-case swizzle, (tho you'll need a `float1` in the single component case above).

Reply:
Changed it